urlfd018a75f7_highlanderkev_StockExVirtuoso_tgz-pJ8-driver_Phase4J8/src/driver/Phase1.java

```diff
@@ -0,0 +1,244 @@
+        } catch (InvalidPriceOperation ex) {
+            System.out.println("FAILED: " + ex.getMessage());
+        }
+        try {
+            Price results = testPriceHolder.get(2).add(testPriceHolder.get(3));
+            System.out.format(format, testPriceHolder.get(2), '+', testPriceHolder.get(3), results, results.toString().equals("$-51.03") ? "CORRECT" : "ERROR");
+        } catch (InvalidPriceOperation ex) {
+            System.out.println("FAILED: " + ex.getMessage());
+        }
+        try {
+            Price results = testPriceHolder.get(3).multiply(4);
+            System.out.format(format, testPriceHolder.get(3), '*', 4, results, results.toString().equals("$1.96") ? "CORRECT" : "ERROR");
+        } catch (InvalidPriceOperation ex) {
+            System.out.println("FAILED: " + ex.getMessage());
+        }
+        try {
+            Price results = testPriceHolder.get(4).subtract(testPriceHolder.get(5));
+            System.out.format(format, testPriceHolder.get(4), '-', testPriceHolder.get(5), results, results.toString().equals("$-12.89") ? "CORRECT" : "ERROR");
+        } catch (InvalidPriceOperation ex) {
+            System.out.println("FAILED: " + ex.getMessage());
+        }
+        try {
+            Price results = testPriceHolder.get(5).add(testPriceHolder.get(6));
+            System.out.format(format, testPriceHolder.get(5), '+', testPriceHolder.get(6), results, results.toString().equals("$102.00") ? "CORRECT" : "ERROR");
+        } catch (InvalidPriceOperation ex) {
+            System.out.println("FAILED: " + ex.getMessage());
+        }
+        try {
+            testPriceHolder.get(8).add(testPriceHolder.get(0));
+            System.out.println("ERROR: Adding a LIMIT price to a MARKET Price succeeded: " + testPriceHolder.get(8) + " + " + testPriceHolder.get(0));
+        } catch (InvalidPriceOperation ex) {
+            System.out.println("CORRECT: " + ex.getMessage() + ": " + testPriceHolder.get(8) + " + " + testPriceHolder.get(0));
+        }
+        try {
+            testPriceHolder.get(8).subtract(testPriceHolder.get(0));
+            System.out.println("ERROR: Subtracting a LIMIT price from a MARKET Price succeeded: " + testPriceHolder.get(8) + " - " + testPriceHolder.get(0));
+
+        } catch (InvalidPriceOperation ex) {
+            System.out.println("CORRECT: " + ex.getMessage() + ": " + testPriceHolder.get(8) + " - " + testPriceHolder.get(0));
+        }
+        try {
+            testPriceHolder.get(8).multiply(10);
+            System.out.println("ERROR: Multiplying a MARKET price succeeded: " + testPriceHolder.get(8) + " + 10");
+
+        } catch (InvalidPriceOperation ex) {
+            System.out.println("CORRECT: " + ex.getMessage() + ": " + testPriceHolder.get(8) + " * 10");
+        }
+        System.out.println();
+    }
+
+    private static void verifyBooleanChecks() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 123-131 into a separate method.